### ISSUE TITLE
Carbs and water fluctuations might confound results.

### ISSUE THREAD

**Issue by ghost** [arielf/weight-loss#8]
Awesome project, as a strength athlete and master in data science, this project includes my two biggest interests! I'm interested in collaborating with this project. Right of the bat I see one major flaw in the way to interpret the results:

Weight alone might not be sufficient to measure the important properties of your body composition. Aside from fat, your weight is also composed of muscle and water. Weight loss in the form of pure muscle would in general not be desired from an aesthetical or health standpoint. Water is an even bigger problem: as carbs are transformed to glycogen and stored in the muscles, for every glycogen particle, 4 water particle are stored around it. Weight spikes after a carb meal are normal and don't have much to do with fat loss. Over the long term water levels stay relatively stable, so comparing high carb weeks with low carb weeks might be a way to cancel this factor out.

**Comment by TylerL-uxai** [comment]
Yes, but wouldn't this be cool to do on a week to week basis using various diets and large sets of people? Just spitballing here. Is there a way we can measure pure fat? What if we use those calipers or something and concentrate on a single area of the body that indicates most significantly fat loss? (I'm thinking stomach). This could be the new science on diet, and finally settle which methods for fat loss are most significant. I applaud his effort.

Edit: What if we have a single day that has constant food prior to the weighing? Example: Every Saturday, eat the same stuff. Weigh on Sunday. And then throughout the week try various diets and exercise.

P.S. A youtube video showing you running the machine learning stuff in R might be extremely helpful to students of machine learning. I took the course online and am still extremely daunted by running my own setup.

**Comment by djui** [comment]
Maybe related: Could you quickly explain why coffee is not in the graph? Do you entirely drink tea/no tea? If not, would you say anything significant about coffee vs tea?

**Comment by arielf** [comment]
@labodyn yes, glycogen and hydration affects weight a lot and they contribute to daily variations which makes it challenging to get good and accurate signal from a small dataset.  This is one of the reasons I added a disclaimer about these particular results.  There's probably a lot of noise and mis-ordering in them.  Once I got into the LCHF routine, the actual long-term weight trend is showing that the loss is independent of daily water retention.

My goal was specifically to lose body fat. This may not be good for everyone. I just believe and feel it was good for me.

@TylerL-uxai -sticking to some routine and weighing once a week can help with accuracy (daily deltas are very small).  Similarly making the sample of people much larger will make the results more significant. 

Now that this code is shared I hope we can get more data and contributions from others.

R course; keep working on it.  I know it takes time to learn, but there's no better way but to learn coding than by doing it.  Start with examples, modify them, write your own, repeat...
